src/gui/executionlogwidget.cpp

```diff
@@ -32,75 +32,45 @@
 #include <QPalette>
 
 #include "base/global.h"
-#include "loglistwidget.h"
+#include "base/logger.h"
+#include "log/logfiltermodel.h"
+#include "log/loglistview.h"
+#include "log/logmodel.h"
 #include "ui_executionlogwidget.h"
 #include "uithememanager.h"
 
-ExecutionLogWidget::ExecutionLogWidget(QWidget *parent, const Log::MsgTypes &types)
+ExecutionLogWidget::ExecutionLogWidget(QWidget *parent, const Log::MsgTypes types)
     : QWidget(parent)
     , m_ui(new Ui::ExecutionLogWidget)
-    , m_msgList(new LogListWidget(MAX_LOG_MESSAGES, types, this))
-    , m_peerList(new LogListWidget(MAX_LOG_MESSAGES, Log::ALL, this))
+    , m_msgFilterModel(new LogFilterModel(types, this))
 {
     m_ui->setupUi(this);
 
+    LogListView *msgList = new LogListView(this);
```

Comment:
move the declaration lower to just where it is needed.

Reply:
done